flow/scripts/generate-variables-docs.py

```diff
@@ -17,10 +17,10 @@
 preferred_order = ["synth", "floorplan", "place", "cts", "grt", "route", "final"]
 stages = {stage for value in data.values() for stage in value.get("stages", [])}
 # convert set of stages to stages in a list in the preferred order, but
-# list all stages
-stages = [stage for stage in preferred_order if stage in stages] + [
-    stage for stage in stages if stage not in preferred_order
-]
+# list all stages and sort the rest for a stable order
+stages = [stage for stage in preferred_order if stage in stages] + sorted(
+    [stage for stage in stages if stage not in preferred_order]
+)
```

Comment:
@vvbandeira I threw in a sorted() here to get a stable order, belt and suspenders. The order is a result of reading in the .yaml and the implementation details of set() in python, so the order is the result of a long line of assumptions and implementation details that I can't account for. A sorted() here puts my mind to ease and may even be necessary.